tests/lib/state_test.py

```diff
@@ -138,3 +140,138 @@ def _extra_state(self):
                 {'name': 'eth1', 'state': 'up', 'type': 'unknown'}
             ]
         })
+
+
+def _get_test_ipv4_routes():
+    return [
+        {
+            Route.DESTINATION: '198.51.100.0/24',
+            Route.METRIC: 103,
+            Route.NEXT_HOP_INTERFACE: 'eth1',
+            Route.NEXT_HOP_ADDRESS: '192.0.2.1',
+            Route.TABLE_ID: 50
+        },
+        {
+            Route.DESTINATION: '203.0.113.0/24',
+            Route.METRIC: 103,
+            Route.NEXT_HOP_INTERFACE: 'eth1',
+            Route.NEXT_HOP_ADDRESS: '192.0.2.2',
+            Route.TABLE_ID: 51
+        }
+    ]
+
+
+def _get_test_ipv6_routes():
+    return [
+        {
+            Route.DESTINATION: '2001:db8:a::/64',
+            Route.METRIC: 103,
+            Route.NEXT_HOP_INTERFACE: 'eth1',
+            Route.NEXT_HOP_ADDRESS: '2001:db8:1::a',
+            Route.TABLE_ID: 50
+        },
+        {
+            Route.DESTINATION: '2001:db8:b::/64',
+            Route.METRIC: 103,
+            Route.NEXT_HOP_INTERFACE: 'eth1',
+            Route.NEXT_HOP_ADDRESS: '2001:db8:1::b',
+            Route.TABLE_ID: 51
+        }
+    ]
+
+
+class TestAssertRouteState(object):
+    parametrize_ip_ver_routes = pytest.mark.parametrize(
+        'routes',
+        [(_get_test_ipv4_routes()),
+         (_get_test_ipv6_routes())],
+        ids=['ipv4', 'ipv6'])
+
+    @parametrize_ip_ver_routes
+    def test_hash_unique(self, routes):
+        assert hash(state._Route(routes[0])) == hash(state._Route(routes[0]))
+
+    @parametrize_ip_ver_routes
+    def test_obj_unique(self, routes):
+        route = routes[0]
+        new_route = copy.deepcopy(routes[0])
+        assert state._Route(route) == state._Route(new_route)
```

Comment:
I'm pretty sure that on py2 this test negative will not pass:
`assert not (state._Route(route) != state._Route(new_route))`

Or just: `assert state._Route(route) != state._Route(different_route)`

Reply:
> Or just: assert state._Route(route) != state._Route(different_route)

Correction, I meant that this will pass: assert state._Route(route) != state._Route(new_route) 
Because it will check for identity instead of equality.

Reply:
The tox passed for running on py27 about the test negative

Reply:
negative test cases added.